File: backend/microservices/courses/src/main/java/com/esprit/microservice/courses/service/instructor/formations/InstructorFormationService.java
package com.esprit.microservice.courses.service.instructor.formations;


import com.esprit.microservice.courses.entity.formations.Formation;

import java.util.List;

public interface InstructorFormationService {

    Formation createFormation(Formation formation, Long instructorId);

    List<Formation> getFormationsByInstructor(Long instructorId);

    Formation getFormationByIdForInstructor(Long formationId, Long instructorId);

    Formation updateFormation(Long formationId, Long instructorId, Formation formation);
}

File: frontend/src/app/core/services/instructor-formation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Formation } from '../models/formation.model';

@Injectable({
  providedIn: 'root'
})
export class InstructorFormationService {
    private apiUrl = '/api/instructor/formations';

  constructor(private http: HttpClient) {}

  getMyFormations(): Observable<Formation[]> {
    return this.http.get<Formation[]>(`${this.apiUrl}/my`);
  }

  getFormationById(formationId: number): Observable<Formation> {
    return this.http.get<Formation>(`${this.apiUrl}/${formationId}`);
  }

  createFormation(payload: Formation): Observable<Formation> {
    return this.http.post<Formation>(this.apiUrl, payload);
  }

  updateFormation(formationId: number, payload: Formation): Observable<Formation> {
    return this.http.put<Formation>(`${this.apiUrl}/${formationId}`, payload);
  }
}